File: reverse-integer/reverse-integer.js
// URL : https://leetcode.com/problems/reverse-integer/
// 116 ms 37.4 MB
var reverse = function(x) {
    let y = 0;
    while(x!=0){
        y = y * 10 + x % 10;
        x = (x - x % 10) / 10;
    }
    return (y > 2147483647 || y < -2147483648)? 0 : y;
};

console.log(reverse(1234))
console.log(reverse(12345))
console.log(reverse(-12345))
console.log(reverse(2147483647))
console.log(reverse(1534236469))
console.log(reverse(-2147483648))

File: reverse-integer/reverse-integer.go
// URL : https://leetcode.com/problems/reverse-integer/
// 0 ms 2.3 MB
package main
import "fmt"
func reverse(x int) int {
    y := 0
    for ; x != 0; x /= 10 {
        y = y * 10 + x % 10
    }

    if y > 2147483647 || y < -2147483648{
        return 0
    }
    return y
}

func main(){
    fmt.Println(reverse(1234))
    fmt.Println(reverse(12345))
    fmt.Println(reverse(-12345))
    fmt.Println(reverse(2147483647))
    fmt.Println(reverse(1534236469))
    fmt.Println(reverse(-2147483648))
}
